compiler/ballerina-treegen/src/main/resources/syntax_tree_descriptor.json

```diff
@@ -2877,7 +2878,7 @@
                 },
                 {
                     "name": "intermediateClauses",
-                    "type": "IntermediateClauseNode",
+                    "type": "ClauseNode",
```

Comment:
Any idea why do we need this change?

Reply:
<https://ballerina.io/swan-lake/learn/by-example/query-expression.html>

In this code, specifically in the `QueryPipelineNode`, `order by student.firstName descending` part is a `OrderByClauseNode` which is extended from `ClauseNode` directly. So, it cannot exists inside the `NodeList<IntermediateClauseNode`.

Here is a small code segment with this problem.

```ballerina
public function main() {
    Report[] reportList = from var student in studentList
       where student.gpa >= 2.0
       order by student.firstName descending
       limit 2
       select {};
}
```

Reply:
I see. 
Issue seems to be in the `OrderByClauseNode`. We should make the `base` type of `OrderByClauseNode` to `IntermediateClauseNode`.